src/wallet/diagnose.h

```diff
@@ -0,0 +1,782 @@
+        } else if (g_nTimeBestReceived > 0 && diff < warn_diff) {
+            m_results_string = "Warning: 80 block difficulty is less than %1. This wallet is probably forked.";
+            std::string ss = argToString(warn_diff);
+            m_results_string_arg.push_back(ss);
+
+            m_results_tip = "Your difficulty is very low and your wallet is probably forked. Please ensure you are "
+                            "running the latest version and try removing the blockchain database and resyncing from "
+                            "genesis using the menu option. (Note this will take 2-4 hours.)";
+            m_results = Diagnose::WARNING;
+        } else {
+            m_results_string = "Passed: 80 block difficulty is %1.";
+            std::string ss = argToString(diff);
+            m_results_string_arg.push_back(ss);
+            m_results = Diagnose::PASS;
+        }
+    }
+};
+/**
+ * check ETTS
+ * This is only checked if wallet is a researcher wallet because the purpose is to
+ * alert the owner that his stake time is too long and therefore there is a chance
+ * of research rewards loss between stakes due to the 180 day limit.
+ */
+class CheckETTS : public Diagnose
+{
+public:
+    CheckETTS()
+    {
+        m_test_name = Diagnose::CheckETTS;
+    }
+    void runCheck()
+    {
+        // This test is only applicable if the wallet is in researcher mode.
+        if (!m_researcher_mode) {
+            m_results = Diagnose::NONE;
+            return;
+        }
+
+        double diff;
+        {
+            LOCK(cs_main);
+            diff = GRC::GetAverageDifficulty(80);
+        }
+
+        double ETTS = GRC::GetEstimatedTimetoStake(true, diff) / (24.0 * 60.0 * 60.0);
+        std::string rounded_ETTS;
+
+        // round appropriately for display.
+        if (ETTS >= 100) {
+            rounded_ETTS = RoundToString(ETTS, 0);
+        } else if (ETTS >= 10) {
+            rounded_ETTS = RoundToString(ETTS, 1);
+        } else {
+            rounded_ETTS = RoundToString(ETTS, 2);
+        }
+
+        if (g_nTimeBestReceived == 0 && OutOfSyncByAge()) {
+            m_results_tip = "Your wallet is not in sync and has not previously been in sync during this run, please "
+                            "wait for the wallet to sync and retest. If there are other failures preventing the "
+                            "wallet from syncing, please correct those items and retest to see if this test passes.";
+            m_results = Diagnose::WARNING;
+        } else {
+            // ETTS of zero actually means no coins, i.e. infinite.
+            if (ETTS == 0.0) {
+                m_results_tip = "You have no balance and will be unable to retrieve your research rewards when solo "
+                                "crunching. You should acquire GRC to stake so you can retrieve your research rewards. "
+                                "Please see https://gridcoin.us/guides/boinc-install.htm.";
```

Comment:
Should there be a note about MRC added here?

Reply:
Can you give me more info about MRC ? What MRC stands for and what is the proposed changes?

Reply:
I resolved it in my PR to yours. :) Do a search of the main repo PRs for MRC. You will find a lot. There is also good info at gridcoin.us. MRC stands for Manual Rewards Claim.